// Copyright 2022 ReductStore
// This Source Code Form is subject to the terms of the Mozilla Public
//    License, v. 2.0. If a copy of the MPL was not distributed with this
//    file, You can obtain one at https://mozilla.org/MPL/2.0/.

syntax = "proto3";

package reduct.proto.api;

// Stats of entry
message EntryInfo {
  string name = 1;          // name of bucket
  uint64 size = 2;          // size of bucket in bytes
  uint64 record_count = 3;  // number of records
  uint64 block_count = 4;   // number of blocks
  uint64 oldest_record = 5; // unix timestamp of oldest record in microseconds
  uint64 latest_record = 6; // unix timestamp of latest record in microseconds
}

// Record
message RecordInfo {
  uint64 ts = 1;      // timestamp of record in microseconds
  uint64 size = 2;    // size in bytes
}

// List of records
message RecordInfoList {
  repeated RecordInfo records = 1;
}

// Query Info
message QueryInfo {
  uint64  id = 1; // id to fetch records
}
